src/workerd/api/node/url.c++

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+#include "url.h"
+
+#include "ada.h"
+
+namespace workerd::api::node {
+kj::ArrayPtr<const char> UrlUtil::domainToASCII(kj::String domain) {
+  if (domain.size() == 0) {
+    return {};
```

Comment:
I wonder if these should return `kj::Maybe<kj::ArrayPtr<const char>>` instead so we can differentiate between a response that produces a zero-length result vs. a response that fails (like in the `!out->set_hostname(...` block below. Not blocking tho

Reply:
Technically yes, but Node.js implementation returns empty string for invalid value. On top of that to_unicode() function in Ada also returns empty string if it's invalid.